### PR TITLE
fix: remove scroll-to-bottom requirement in redesigned transaction confirmations

### PR DESCRIPTION
**Pull request by vinistevam**
## **Description**
This PR addresses the removal of the scroll-to-bottom requirement in the redesigned transaction confirmation screens. It eliminates the need for users to scroll to the bottom in order to enable the confirm button, streamlining the confirmation process. The scroll-to-bottom arrow is also removed, ensuring a smoother user experience without unnecessary interaction barriers.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27910?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/3495

## **Manual testing steps**

1. Go to test dapp
2. Have 2+ transaction insights snaps installed
3. Click Create Token
4. See the confirm button disabled until you scroll to bottom 

## **Screenshots/Recordings**

[deploy.webm](https://github.com/user-attachments/assets/79716a68-e70f-456a-b962-ccec8732935b)

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/pages/confirmations/components/confirm/scroll-to-bottom/scroll-to-bottom.tsx [hunk 1]
```diff
@@ -71,6 +80,11 @@ const ScrollToBottom = ({ children }: ContentProps) => {
   }, [currentConfirmation?.id, previousId, ref?.current]);
 
   useEffect(() => {
+    if (isTransactionRedesign) {
+      setIsScrollToBottomCompleted(true);
+      return;
+    }
+
     setIsScrollToBottomCompleted(!isScrollable || hasScrolledToBottom);
   }, [isScrollable, hasScrolledToBottom]);
```
Comment (OGPoyraz) [inline ui/pages/confirmations/components/confirm/scroll-to-bottom/scroll-to-bottom.tsx:89]: Should this hook to have `isTransactionRedesign` dependency?

File: ui/pages/confirmations/components/confirm/scroll-to-bottom/scroll-to-bottom.tsx [hunk 2]
```diff
@@ -49,6 +51,13 @@ const ScrollToBottom = ({ children }: ContentProps) => {
     offsetPxFromBottom: 0,
   });
 
+  const isTransactionRedesign = REDESIGN_USER_TRANSACTION_TYPES.includes(
```
Comment (OGPoyraz) [inline ui/pages/confirmations/components/confirm/scroll-to-bottom/scroll-to-bottom.tsx:54]: Maybe a dumb question, should this to be only check `REDESIGN_USER_TRANSACTION_TYPES` but also if user opted-in for redesigned transactions?

Comment (vinistevam) [reply ui/pages/confirmations/components/confirm/scroll-to-bottom/scroll-to-bottom.tsx:54]: I assumed `ScrollToBottom` is used only for redesigned confirmations. I will double-check.

File: ui/pages/confirmations/components/confirm/scroll-to-bottom/scroll-to-bottom.tsx (hunk 1 above)
Comment (vinistevam) [reply ui/pages/confirmations/components/confirm/scroll-to-bottom/scroll-to-bottom.tsx]: Great catch, [done](https://github.com/MetaMask/metamask-extension/pull/27910/commits/a1a3614aa5e12edc8045f399658ee84726e85de3).

File: ui/pages/confirmations/components/confirm/scroll-to-bottom/scroll-to-bottom.tsx (hunk 2 above)
Comment (OGPoyraz) [reply ui/pages/confirmations/components/confirm/scroll-to-bottom/scroll-to-bottom.tsx:54]: That make sense if that's the case

**Review by OGPoyraz — DISMISSED** [review]

File: ui/pages/confirmations/components/confirm/scroll-to-bottom/scroll-to-bottom.tsx (hunk 2 above)
Comment (pedronfigueiredo) [reply ui/pages/confirmations/components/confirm/scroll-to-bottom/scroll-to-bottom.tsx:54]: In this case we probably want to use `REDESIGN_DEV_TRANSACTION_TYPES` instead, which is a superset of `REDESIGN_USER_TRANSACTION_TYPES` and includes any types not yet visible to the user but already in active development

**Review by pedronfigueiredo — DISMISSED** [review]

**Review by jpuri — DISMISSED** [review]

**Review by OGPoyraz — APPROVED** [review]

**Review by digiwand — APPROVED** [review]

**Pull request merged**